### PR TITLE
Fixes to api_v2 plus fully containerized solution for quads

### PR DESCRIPTION
**Pull request by grafuls**
This PR contains the following enhancements:
* Fully containerized environment with individual containers for (docker-compose file included/details on README):
  - nginx
  - php-fpm/wordpress
  - mariadb (wordpress db)
  - quads
  - mongodb (quads db)
* Fixes to:
  - api_v2 entrypoints and several quads-cli commands
* New api_v2 entrypoints:
  - /api/v2/cloud?id=5c2c91b796ef1c00083ae015
  - /api/v2/host?id=5c2c91df96ef1c00083ae01b
  - /api/v2/cloud?name=cloud01
  - /api/v2/host?name=host01.example.com
  - /api/v2/current_schedule?date=2018-01-01T22:00:00
  - /api/v2/moves (WIP)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -230,6 +232,43 @@ systemctl start quads-daemon.service
 ```
 
    - Note: RPM installations will have ```quads-cli``` and tools in your system $PATH but you will need to login to a new shell to pick it up.
+   
+### Running Quads with Docker compose (Experimental)
+   - Clone repo and move to docker directory
+```bash
+git clone https://github.com/grafuls/quads.git && cd quads/docker
```
Comment (sadsfae) [inline README.md:239]: Do we want to just use the normal QUADS url here so it won't need to be changed later?

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    build: .
+    container_name: quads
+    ports:
+      - 8080:8080
+    environment:
+      - MONGODB_IP=quads-db
+      - WORDPRESS_IP=wordpress
+
+  quads-db:
+    image: mongo:4.0.4-xenial
```
Comment (sadsfae) [inline docker/docker-compose.yml:59]: What do you think about using an Alpine image instead for mongo?  Probably lighter-weight  https://github.com/mvertes/docker-alpine-mongo

File: README.md (hunk 1 above)
Comment (grafuls) [reply README.md:239]: Absolutely, missed that

File: docker/etc/wordpress/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+			set_config '$table_prefix' "$WORDPRESS_TABLE_PREFIX"
+		fi
+
+		if [ "$WORDPRESS_DEBUG" ]; then
+			set_config 'WP_DEBUG' 1 boolean
+		fi
+
+        cp -r /var/tmp/wp-themes/quads /var/www/html/wp-content/themes/
+        unzip -n /var/tmp/wp-themes/krusze.0.9.7.zip -d /var/www/html/wp-content/themes/
+        unzip -n /var/tmp/wp-plugins/contact-bank.zip -d /var/www/html/wp-content/plugins/
```
Comment (sadsfae) [inline docker/etc/wordpress/entrypoint.sh:234]: Nice, pulling in our plugins too :+1:

File: docker/docker-compose.yml (hunk 2 above)
Comment (grafuls) [reply docker/docker-compose.yml:59]: Good point, not sure if the bindings will be the same but I'll check

File: docker/etc/wordpress/themes/quads/style.css [hunk 4]
```diff
@@ -0,0 +1,10 @@
+/*
+Theme Name: Quads
+Description: Quads ultimate theme
```
Comment (sadsfae) [inline docker/etc/wordpress/themes/quads/style.css:3]: ULTIMATE!!!!!!1

**Review by sadsfae — COMMENTED** [review]
Looks awesome, an incredible amount of work.  I just had some minor questions/comments.

File: docker/docker-compose.yml (hunk 2 above)
Comment (grafuls) [reply docker/docker-compose.yml:59]: Come to think about it, this is not an official image provided by the Mongo team, for security reasons I rather stick with the official xenial, though would be a nice improvement.
Image sizes:
mvertes/alpine-mongo:latest     124MB
mongo:4.0.4-xenial                    383MB

**Pull request merged**